lock_api/src/remutex.rs

```diff
@@ -178,6 +183,9 @@ unsafe impl<R: RawMutex + Send, G: GetThreadId + Send, T: ?Sized + Send> Send
     for ReentrantMutex<R, G, T>
 {
 }
+// REVIEW: this naively seems like it should require `T: Sync` as well because
+// if you share `ReentrantMutex<R, G, T>` across threads then you also have
+// access to `&T` across all threads. I may be getting this backwards though.
```

Comment:
Nice catch! This seems to be a typo.

Reply:
If we add that requirement then a `ReentrantMutex<RefCell<T>>` does not become `Sync` any more. And the documentation for `ReentrantMutex` explicitly states that one should use `RefCell` to get mutable access to the data in the mutex. The `is_mutex` test in this file breaks.